src/Toolkit/V.php

```diff
@@ -248,6 +248,21 @@ public static function __callStatic(string $method, array $arguments): bool
                 $date['warning_count'] === 0;
     },
 
+    /**
+     * Parses the first two parameters as dates and then
+     * validates them as specified by the third parameter
+     */
+    'dates' => function ($value, $test, $validator): bool {
+        $value = strtotime($value);
+        $test  = strtotime($test);
+
+        if (is_int($value) === true && is_int($test) === true) {
+            return V::$validator($value, $test);
```

Comment:
The call stack that is behind this line can become quite deep (e.g. `max` calls `V::__call()`, which runs the `size` validator, which calls `is_numeric`) considering that the validator is going to be run on every collection element and considering that the actual comparison is very simple.

So my suggestion would be to use a simple `switch` here with `$operator` options like `>=` (like in the `size` validator) and direct comparison without calling other methods. Another advantage besides being faster: Much, much easier to read and to understand without having to follow the call stack. And because the code is so simple, the benefit of making it DRY is negligible compared to the benefits of using a `switch` directly.

Reply:
Ok sorry, now I see why you changed it to `$operator` :D

Hmm, makes the method a bit less flexible, but on the other hand it would be in line with the `size` validator, I like that.